provision-osx/tasks/configure_buildfarm_proxy.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+-
+  name: Set node proxy configuration with authentication
+  shell: "networksetup -{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }} on {{ proxy_user }} {{ proxy_pass }}"
+  with_items: "{{ proxy_ctx }}"
+  become: true
+  when: 
+    - proxy_host is defined and proxy_host != ''
+    - proxy_user is defined and proxy_user != ''
+    - proxy_pass is defined and proxy_pass != ''
+
+-
+  name: Set node proxy configuration without authentication
+  shell: "networksetup -{{ item }} {{ proxy_device }} {{ proxy_host }} {{ proxy_port }}"
+  with_items: "{{ proxy_ctx }}"
+  become: true
+  when:
+    - proxy_host is defined and proxy_host != ''
+    - proxy_user is not defined or proxy_user == ''
+    - proxy_pass is not defined or proxy_pass == ''
+
+-
+  name: Sets proxy status
+  shell: "networksetup -set{{ item }}state {{ proxy_device }} {{ 'on' if proxy_host is defined and proxy_host != '' else 'off'}}"
```

Comment:
I am not sure this is the most elegant way of doing this, is there a better way?